criu/arch/x86/cpu.c

```diff
@@ -261,25 +262,37 @@ static int cpu_validate_features(compel_cpuinfo_t *cpu_info)
 #undef __mismatch_fpu_bit
 
 		/*
-		 * Make sure the xsave features are compatible. We already hit the
-		 * issue with libc where we've checkpointed the container on old
-		 * machine but restored on more modern one and libc fetched new
-		 * xsave frame size directly by xsave instruction with greedy
-		 * feature mask causing programs to misbehave.
+		 * Make sure the xsave features are compatible. We were relying
+		 * on our guess about hardware "backward compatibility" and
+		 * used ">" check here for a long time. But it looks better to
+		 * explicitly check that all xfeature bits from source are
+		 * present on the destination.
 		 */
-		if (cpu_info->xfeatures_mask > rt_cpu_info.xfeatures_mask) {
-			uint64_t m = cpu_info->xfeatures_mask & ~rt_cpu_info.xfeatures_mask;
+		if ((m = cpu_info->xfeatures_mask & ~rt_cpu_info.xfeatures_mask)) {
 			pr_err("CPU xfeatures has unsupported bits (%#llx)\n",
 			       (unsigned long long)m);
```

Comment:
While at here, is it possible to print it with `"...%#"PRIx64")\n"`, removing needless cast?

Reply:
done.